include/envoy/stats/stats.h

```diff
@@ -194,6 +196,11 @@ class Sink {
    */
   virtual void flushGauge(const Gauge& gauge, uint64_t value) PURE;
 
+  /**
+   * Flush all Histograms. This will be called before endFlush().
+   */
+  virtual std::list<HistogramStatistics> flushHistograms() PURE;
```

Comment:
Why does this return a list of histograms? Can we expand the docs here? It seems like flushing should be orthogonal to getting the current known histograms.

Reply:
This is some what related to Store comment above. I have added this because Server does not have Sink instance to get it, so at the flushHistograms I am returning so that server can  get the current known histograms. However if we can get it from Store we may not need this. Let me think from that perspective and see if we can move this to store.